app/src/main/java/jp/co/soramitsu/app/root/presentation/RootViewModel.kt

```diff
@@ -19,19 +26,44 @@ class RootViewModel(
     private val rootRouter: RootRouter,
     private val externalConnectionRequirementFlow: MutableStateFlow<ExternalRequirement>,
     private val resourceManager: ResourceManager,
-    private val networkStateMixin: NetworkStateMixin
+    private val networkStateMixin: NetworkStateMixin,
+    private val getAppVersion: GetAppVersion,
 ) : BaseViewModel(), NetworkStateUi by networkStateMixin {
+    companion object {
+        private const val IDLE_MINUTES: Long = 20
+    }
 
     private var willBeClearedForLanguageChange = false
+    private var timeInBackground: Date? = null
+
+    private val _showUnsupportedAppVersionAlert = MutableLiveData<Event<Unit>>()
+    val showUnsupportedAppVersionAlert: LiveData<Event<Unit>> = _showUnsupportedAppVersionAlert
+
+    private val _openPlayMarket = MutableLiveData<Event<Unit>>()
+    val openPlayMarket: LiveData<Event<Unit>> = _openPlayMarket
+
+    private val _closeApp = MutableLiveData<Event<Unit>>()
+    val closeApp: LiveData<Event<Unit>> = _closeApp
 
     init {
+        checkAppVersion()
         interactor.runBalancesUpdate()
             .onEach { handleUpdatesSideEffect(it) }
             .launchIn(this)
 
         updatePhishingAddresses()
     }
 
+    private fun checkAppVersion() = viewModelScope.launch {
+        val appVersion = getAppVersion().replace(".", "")
```

Comment:
why replace "."? 2.1.11 and 2.11.1 will have no difference

Reply:
right, will fix